src/core/lib/surface/server.cc

```diff
@@ -1161,6 +1161,11 @@ void grpc_server_setup_transport(grpc_server* s, grpc_transport* transport,
 
   channel = grpc_channel_create(nullptr, args, GRPC_SERVER_CHANNEL, transport,
                                 resource_user);
+  if (channel == nullptr) {
+    gpr_log(GPR_ERROR, "Error creating channel. Will not setup transport");
+    grpc_transport_destroy(transport);
+    return;
+  }
```

Comment:
I think we at least need a comment in the grpc_server_setup_transport to say the transport will be destroyed on error. Ideally we should return an error from here indicating what has happened. If some transport implements an in-place transport destroy and then try to reference the transport after this function returns. This will cause problems.